server/src/main/java/org/elasticsearch/action/DocWriteRequest.java

```diff
@@ -161,7 +161,12 @@ public interface DocWriteRequest<T> extends IndicesRequest, Accountable {
     /**
      * Finalize the request before executing or routing it.
      */
-    void process(IndexRouting indexRouting);
```

Comment:
Let's undo this change to `DocWriteRequest` and instead add IndexRequest as a parameter to `IndexRouting#indexShard(...)` method. I think this should be a simpler change.

 Or maybe add a `Consumer` parameter that then would provide the routing hash to IndexRequest.

Reply:
Yeah a consumer looks better, good point.